tests/web/test_browse_folder_route.py

```
@@ -0,0 +1,233 @@
+            patch("file_organizer.api.routers.setup.sys.platform", "darwin"),
+            patch(
+                "file_organizer.api.routers.setup.subprocess.run",
+                return_value=mock_result,
+            ),
+        ):
+            resp = client.get("/api/setup/browse-folder")
+        assert resp.json()["path"] == "/Users/rahul/Desktop/"
+
+    def test_calls_osascript_posix_path(self, client: TestClient) -> None:
+        """Must call osascript with 'POSIX path of (choose folder)'."""
+        mock_result = self._make_run_result("/tmp/\n")
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/\n"

(S108)